CompetitionMARS/Tests/CertificationTests.cs

```diff
@@ -0,0 +1,364 @@
+           
+            test.Log(Status.Pass, "success");
+            string folderPath = Path.Combine(Directory.GetCurrentDirectory(), "Screenshots");
+            Directory.CreateDirectory(folderPath);
+
+            // File name with timestamp
+            string screenshotPath = Path.Combine(folderPath, $"test_{DateTime.Now:yyyyMMdd_HHmmss}.png");
+
+            // Take screenshot
+            Screenshot screenshot = ((ITakesScreenshot)driver).GetScreenshot();
+            screenshot.SaveAsFile(screenshotPath);
+            test.AddScreenCaptureFromPath(screenshotPath);
+            test.Pass("Screenshot added to Extent Report");
+
+            // Assert that the certification was successfully added
+            bool isCertificationAdded = certPage.IsCertificationAdded(cert.CertificateOrAward);
+            Assert.IsTrue(isCertificationAdded, "Certification entry was not added successfully.");
+            if (true)
+            {
+                test.Log(Status.Pass, "success");
+            }
+            else
+            {
+                test.Log(Status.Fail, "failed");
+            }
+            Thread.Sleep(3000);
+
+        }
+        [Test, Order(2)]
+        public void CertificationTest_Invalid()
+        {
+            ExtentTest test = extent.CreateTest("CertificationTest_Invalid");
+
+            CertificationData cert = certPage.GetCertificationData("invaliddata");
+
+            certPage.AddNewCertification(cert.CertificateOrAward, cert.CertifiedFrom, cert.CertificationYear);
+            // Track the added CertificateName for cleanup
+            //addedCertificationsEntries.Add(cert.CertificateOrAward);
+            bool isEntryPresent = certPage.IsCertificationEntryPresent(cert.CertificateOrAward);
+            Assert.IsTrue(isEntryPresent, $"Invalid entry '{cert.CertificateOrAward}' should not be added.");
+            if (true)
+            {
+                test.Log(Status.Pass, "success");
+            }
+            else
+            {
+                test.Log(Status.Fail, "failed");
+            }
+
+        }
+        [Test, Order(3)]
+        public void CertificationTest_MissingField()
+        {
+            ExtentTest test = extent.CreateTest("CertificationTest_MissingField");
+            Certification certPage = new Certification();
+            CertificationData cert = certPage.GetCertificationData("validdata12");
+
+            certPage.AddNewCertification_missingField(cert.CertificateOrAward, cert.CertificationYear);
+            // Assert that an appropriate error message is displayed
+            var errorMessage = certPage.GetErrorMessage();
+            Assert.IsFalse(errorMessage.Contains("Please enter Certification Name,Certification From and Certification Year"),
+                "Expected error message for missing field is not displayed.");
+            IWebElement cancelButton = driver.FindElement(By.XPath("//input[contains(@value,'Cancel')]"));
```

Comment:
your test class should not contain any web elements, they should all be in pages! tests should only contain steps

Reply:
if you need something from the pages to assert then you should create methods there and return the value